crates/neon/src/sys/async_work.rs

```diff
@@ -150,7 +150,9 @@ unsafe extern "C" fn call_complete<I, O, D>(env: Env, status: napi::Status, data
         ..
     } = *Box::<Data<I, O, D>>::from_raw(data.cast());
 
-    napi::delete_async_work(env, work);
+    if napi::delete_async_work(env, work).is_err() {
```

Comment:
I could go either way. We've already unboxed the closure, so we aren't going to leak memory by panicking here. However, it might be a bit more consistent to debug assert.